google-cloud-spanner/README.md

```diff
@@ -2,9 +2,24 @@
 
 [Google Cloud Spanner API](https://cloud.google.com/spanner/) ([docs](https://cloud.google.com/spanner/docs)) provides a fully managed, mission-critical, relational database service that offers transactional consistency at global scale, schemas, SQL (ANSI 2011 with extensions), and automatic, synchronous replication for high availability.
 
-- [google-cloud-spanner API documentation](https://googleapis.dev/ruby/google-cloud-spanner/latest)
-- [google-cloud-spanner on RubyGems](https://rubygems.org/gems/google-cloud-spanner)
-- [Google Cloud Spanner API documentation](https://cloud.google.com/spanner/docs)
+- [google-cloud-spanner API
+  documentation](https://googleapis.dev/ruby/google-cloud-spanner/latest)
+- [google-cloud-spanner on
+  RubyGems](https://rubygems.org/gems/google-cloud-spanner)
+- [Google Cloud Spanner API
+  documentation](https://cloud.google.com/spanner/docs)
+
+## NOTICE: Freezing development of `database` and `instance` classes.
```

Comment:
Do you want to deprecate the methods and classes subject to the freeze? You can add the YARD [`@deprecated`](https://www.rubydoc.info/gems/yard/file/docs/Tags.md#deprecated) tag to the inline docs for each with a description pointing the user to its equivalent in the generated client.

Reply:
That's a good question. It depends on what it means in Ruby to add a deprecated tag. Is it going to add lots of log messages or something when you use that method? I added some and ran the unit tests but didn't see anything even in --verbose mode.

I think we want to make this as pain free as possible for users to continue to use the existing classes if that's what they want. Also by adding a deprecation tag, users may think we plan to delete the classes but we won't be doing that.

So yeah, I'm a bit undecided on this.

Reply:
Having said that, I did add deprecation notices where possible. At a method level in `Project` and at a class level in others.

Reply:
>  Is it going to add lots of log messages or something when you use that method?

It just adds formatted output to the docs, ideally with a link to the new method/class.

https://www.rubydoc.info/gems/yard/file/docs/Tags.md#deprecated

It shouldn't annoy users at all, but it will guide those who read it to the current best usage.